storage/innobase/handler/ha_innodb.cc

```diff
@@ -19054,6 +19057,25 @@ wsrep_innobase_kill_one_trx(
 	DBUG_RETURN(0);
 }
 
+static
+void
+wsrep_wait_until_initialized(
+/*====================*/
+       handlerton*)
+{
+	extern bool trx_rollback_is_active;
+	DBUG_ENTER("wsrep_innodb_wait_untill_initialized");
+
+	WSREP_DEBUG("waiitng until innodb is initialized %d",
+		   trx_rollback_is_active);
+	while (trx_rollback_is_active) {
+		WSREP_DEBUG("innodb rollbacker still active");
+		sleep(1);
+	}
+	WSREP_DEBUG("innodb is now initialized %d", trx_rollback_is_active);
```

Comment:
Please try to avoid adding `/*====*/` comments or splitting the function signature into multiple lines.

There is a typo in the `DBUG_ENTER` comment, and the last `WSREP_DEBUG` message looks superfluous.

Please add a test that initiates server shutdown while a huge transaction is being rolled back. In that case, `trx_rollback_all_recovered()` should terminate without completing the rollback.

What about transactions that are recovered in `XA PREPARE` state? Those will continue to hold table IX locks and implicit record X locks.